returning optional "multi-value" produces invalid generated C code

**V doctor** 
$ v doctor
```
OS: linux, Ubuntu 22.04.1 LTS
Processor: 16 cpus, 64bit, little endian, AMD Ryzen 7 3700X 8-Core Processor
CC version: cc (Ubuntu 11.2.0-19ubuntu1) 11.2.0

getwd: /home/dominik
vmodules: /home/dominik/.vmodules
vroot: /home/dominik/Projects/vlang
vexe: /home/dominik/Projects/vlang/v
vexe mtime: 2022-10-11 12:11:19
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.1 dab1c8a.34f233c

Git version: git version 2.34.1
Git vroot status: weekly.2022.40-47-g34f233c9 (3 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3
```

**What did you do?**

Issue #15989 was fixed hence I tried to adapt code to leverage it. Unfortunately I got another issue.

I tried 
```
struct Aa {
	x string
}
struct Bb {
	a int
}

fn give(succ Aa) ?(Aa, Bb) {
	return match succ.x { 
		"x" {
			succ, Bb{}
		}
		else {
			error("nok")
		}
	}
}

fn main() {	
	res, _ := give(Aa{x:"x"}) or {
		panic("got unexpected err")
	}
	
	assert res.x == "x"
	println('success')
}
```
`v -v -cg -keepc -cc gcc vbug4.v && ./vbug4`

**What did you expect to see?**

`success` on console

**What did you see instead?**

```
> C compiler cmd: 'gcc' '@/tmp/v_1000/vbug4.tmp.c.rsp'
77.125   ms C gcc
/tmp/v_1000/vbug4.tmp.c: In function ‘main__give’:
/tmp/v_1000/vbug4.tmp.c:12095:105: error: ‘_option_multi_return_main__Aa_main__Bb’ has no member named ‘arg0’
12095 |                 opt_ok2(&(multi_return_main__Aa_main__Bb[]) { (_option_multi_return_main__Aa_main__Bb){.arg0=succ,.arg1=((main__Bb){.a = 0,})} }, (_option*)(&_t2), sizeof(multi_return_main__Aa_main__Bb));
      |                                                                                                         ^~~~
/tmp/v_1000/vbug4.tmp.c:12095:110: error: incompatible types when initializing type ‘unsigned char’ using type ‘main__Aa’
12095 |                 opt_ok2(&(multi_return_main__Aa_main__Bb[]) { (_option_multi_return_main__Aa_main__Bb){.arg0=succ,.arg1=((main__Bb){.a = 0,})} }, (_option*)(&_t2), sizeof(multi_return_main__Aa_main__Bb));
      |                                                                                                              ^~~~
/tmp/v_1000/vbug4.tmp.c:12095:116: error: ‘_option_multi_return_main__Aa_main__Bb’ has no member named ‘arg1’
12095 |                 opt_ok2(&(multi_return_main__Aa_main__Bb[]) { (_option_multi_return_main__Aa_main__Bb){.arg0=succ,.arg1=((main__Bb){.a = 0,})} }, (_option*)(&_t2), sizeof(multi_return_main__Aa_main__Bb));
      |                                                                                                                    ^~~~
builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4166,11 +4166,11 @@ fn (mut g Gen) cast_expr(node ast.CastExpr) {
 }
 
 fn (mut g Gen) concat_expr(node ast.ConcatExpr) {
-	mut styp := g.typ(node.return_type)
+	mut styp := g.typ(node.return_type.clear_flag(.optional).clear_flag(.result))
 	if g.inside_return {
-		styp = g.typ(g.fn_decl.return_type)
+		styp = g.typ(g.fn_decl.return_type.clear_flag(.optional).clear_flag(.result))
 	} else if g.inside_or_block {
-		styp = g.typ(g.or_expr_return_type)
+		styp = g.typ(g.or_expr_return_type.clear_flag(.optional).clear_flag(.result))
 	}
 	sym := g.table.sym(node.return_type)
 	is_multi := sym.kind == .multi_return

--- a/vlib/v/tests/fn_with_opt_or_res_of_multi_return_test.v
+++ b/vlib/v/tests/fn_with_opt_or_res_of_multi_return_test.v
@@ -1,0 +1,25 @@
+struct Aa {
+	x string
+}
+
+struct Bb {
+	a int
+}
+
+fn give(succ Aa) ?(Aa, Bb) {
+	return match succ.x {
+		'x' {
+			succ, Bb{}
+		}
+		else {
+			error('nok')
+		}
+	}
+}
+
+fn test_fn_with_opt_of_multi_return() {
+	res, _ := give(Aa{ x: 'x' }) or { panic('got unexpected err') }
+
+	assert res.x == 'x'
+	println('success')
+}

